spec.md

```diff
@@ -180,6 +180,18 @@ contributors is below:
 - [`@tdb-alcorn`](https://github.com/tdb-alcorn)
 - [`@youngar`](https://github.com/youngar)
 
+# File Preamble
+
+A firrtl file includes a magic string and version identifier indicating the 
+version of this standard the file conforms to.  This will not be present on 
+files generated according to versions of this standard prior to the first 
+versioned release of this standard to include this preamble.
+
```

Comment:
does this have to be the first thing in the file, or can it occur anywhere (the term "preamble" implies that it should be the first thing)? Is the "magic string" `firrtl` or is it `firrtl version`? Should this get added to the language reference at the end of the doc?

Reply:
also why do we need to call this a "magic string". What is magic about it, vs just saying the version is specified in this way

Reply:
magic string being an in-data method of determining file type from contents, not extension.  This is used by file on unix (source of the term magic, since that's the file containing the patterns) and for identifying files for mime types.  Since we are adding a version anyway, having a header which can be used for this purpose adds little costs, only 7 bytes, and makes the format a tiny bit more self-identifying.
https://linux.die.net/man/1/file
https://en.wikipedia.org/wiki/List_of_file_signatures

Reply:
cool, thanks for the explanation